gcloud-java-dns/src/main/java/com/google/gcloud/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1426 @@
+          // we do not add this, just note that there would be more and there should be a token
+          hasMorePages = true;
+          break;
+        } else {
+          lastChangeId = change.getId();
+          try {
+            serializedResults.addLast(jsonFactory.toString(
+                OptionParsersAndExtractors.extractFields(change, fields)));
+          } catch (IOException e) {
+            return Error.INTERNAL_ERROR.response(String.format(
+                "Error when serializing change %s in managed zone %s in project %s",
+                change.getId(), zoneName, projectId));
+          }
+        }
+      }
+
+      // either we are listing already, or we check if we should start in the next iteration
+      listing = change.getId().equals(pageToken) || listing;
+      sizeReached = (maxResults != null) && maxResults.equals(serializedResults.size());
+    }
+    boolean includePageToken =
+        hasMorePages && (fields == null || ImmutableList.copyOf(fields).contains("nextPageToken"));
+    return toListResponse(serializedResults, "changes", lastChangeId, includePageToken);
+  }
+
+  /**
+   * Validates a zone to be created.
+   */
+  static Response checkZone(ManagedZone zone) {
+    if (zone.getName() == null) {
+      return Error.REQUIRED.response(
+          "The 'entity.managedZone.name' parameter is required but was missing.");
+    }
+    if (zone.getDnsName() == null) {
+      return Error.REQUIRED.response(
+          "The 'entity.managedZone.dnsName' parameter is required but was missing.");
+    }
+    if (zone.getDescription() == null) {
+      return Error.REQUIRED.response(
+          "The 'entity.managedZone.description' parameter is required but was missing.");
+    }
+    try {
+      int number = Integer.valueOf(zone.getName());
+      return Error.INVALID.response(
+          String.format("Invalid value for 'entity.managedZone.name': '%s'", number));
+    } catch (NumberFormatException ex) {
+      // expected
+    }
+    if (zone.getName().isEmpty()) {
+      return Error.INVALID.response(
+          String.format("Invalid value for 'entity.managedZone.name': '%s'", zone.getName()));
+    }
+    if (zone.getDnsName().isEmpty() || !zone.getDnsName().endsWith(".")) {
+      return Error.INVALID.response(
+          String.format("Invalid value for 'entity.managedZone.dnsName': '%s'", zone.getDnsName()));
+    }
+    TreeSet<String> forbidden = Sets.newTreeSet(
+        ImmutableList.of("google.com.", "com.", "example.com.", "net.", "org."));
+    if (forbidden.contains(zone.getDnsName())) {
+      return Error.NOT_AVAILABLE.response(String.format(
+          "The '%s' managed zone is not available to be created.", zone.getDnsName()));
+    }
+    return null;
+  }
+
+  /**
+   * Validates a change to be created.
+   */
+  static Response checkChange(Change change, ZoneContainer zone) {
+    checkNotNull(zone);
+    if ((change.getDeletions() != null && change.getDeletions().size() > 0)
```

Comment:
Since one of these branches doesn't do anything, you could check for the opposite and remove the else.